### DISCUSSION TITLE
Events for PackerCompile

### DISCUSSION THREAD

**Discussion by axieax** [rebelot/heirline.nvim#76] [Q&A] [answered]
For some reason my LSP clients component breaks when I `:PackerCompile` due to the events in `updates`. Is there a way to resolve this? I've tried wrapping the code in `init` in a `vim.schedule` call already, but that didn't seem to fix this issue.

```lua
{ -- LSP Clients
  static = {
    ignored_clients = { "null-ls", "copilot" },
  },
  -- NOTE: PackerCompile breaks this
  update = { "BufEnter", "BufLeave", "LspAttach", "LspDetach" },
  init = function(self)
    self.clients = vim.tbl_map(function(client)
      if not vim.tbl_contains(self.ignored_clients, client.name) then
        return client.name
      end
    end, vim.lsp.buf_get_clients(0))
  end,
  provider = function(self)
    return table.concat(self.clients, ", ")
  end,
  on_click = {
    name = "LSP Info",
    callback = function()
      vim.api.nvim_command("LspInfo")
    end,
  },
},
```

**Comment by rebelot** [comment, marked as answer]
could you please be more specific on the error you get? I can't see how the update field could break packer.

**Reply by axieax** [reply]
I've resolved this issue now - turns out it was due to my use of `vim.tbl_map`. I did a manual for loop instead and it works now! The issue was just that `self.clients` becomes empty. Not sure why this is the case (happened for galaxyline.nvim as well)...
